### PR TITLE
add meta to bwa index

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/bwa/aln/meta.yml [hunk 1]
```diff
@@ -30,6 +30,11 @@ input:
       description: |
         List of input FastQ files of size 1 and 2 for single-end and paired-end data,
         respectively.
+  - meta:
```
Comment (jfy133) [inline modules/nf-core/bwa/aln/meta.yml:33]: Shouldn't this be meta2?

File: tests/modules/nf-core/bwa/aln/main.nf [hunk 2]
```diff
@@ -34,6 +34,6 @@ workflow test_bwa_aln_paired_end {
     ]
     fasta = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
 
-    BWA_INDEX ( fasta )
+    BWA_INDEX ( [ [:], fasta ] )
```
Comment (maxulysse) [inline tests/modules/nf-core/bwa/aln/main.nf:37]: no nice meta map, like you did in the other PR?

Comment (matthdsm) [reply tests/modules/nf-core/bwa/aln/main.nf:37]: I was lazy and copy pasted from BWA_MEM2 😊

**Review by jfy133 — APPROVED** [review]
LGTM with passing tests 👍

File: tests/modules/nf-core/bwa/aln/main.nf (hunk 2 above)
Comment (matthdsm) [reply tests/modules/nf-core/bwa/aln/main.nf]: There, fixed it for you!

File: modules/nf-core/bwa/index/main.nf [hunk 3]
```diff
@@ -8,11 +8,11 @@ process BWA_INDEX {
         'quay.io/biocontainers/bwa:0.7.17--hed695b0_7' }"
 
     input:
-    path fasta
+    tuple val(meta), path(fasta)
 
     output:
-    path "bwa"         , emit: index
-    path "versions.yml", emit: versions
+    tuple val(meta), path(bwa) , emit: index
```
Comment (maxulysse) [inline modules/nf-core/bwa/index/main.nf:14]:
```suggestion
    tuple val(meta), path("bwa") , emit: index
```

**Comment by maxulysse** [comment]
We need to update tests/subworkflows/nf-core/fgbio_create_umi_consensus as well

**Comment by matthdsm** [comment]
There seem to be a lot of issues with the fgbio subwf.. how did this ever pass CI?

**Comment by matthdsm** [comment]
no wonder... it didn't pass CI, but is a leftover subworkflow from the pre hackathon era...

**Comment by maxulysse** [comment]
I summon @lescai the great

**Comment by lescai** [comment]
sorry guys, several modules have been updated recently and they're breaking changes to the sbw.
I'm waiting for the last of new fgbio modules to pass review and will start working on a revision of the whole workflow. that will obviously work with all new modules.

**Comment by lescai** [comment]
@matthdsm the subworkflow was actually pushed during a hackathon :)

**Comment by matthdsm** [comment]
> @matthdsm the subworkflow was actually pushed during a hackathon :)

Weird, the latest commit is from 10 days ago according to git

**Comment by lescai** [comment]
if the subworkflow is creating these tests problems, perhaps we can remove it straight away (let's call it retire? :P) and I'll submit an updated version anyway as soon as possible?

**Comment by matthdsm** [comment]
meh 🤷🏻 I'm already put in the work to fix it and I'm mostly done. Let's say I fix the current version and you take it from there 😉 I already opened an issue with some stuff that I think needs changing

**Comment by matthdsm** [comment]
Conda gonna conda again, but the rest seem to pass!
@maxulysse, can I get the second ✅ ?

**Comment by matthdsm** [comment]
🎉  Finally!

**Comment by lescai** [comment]
can someone remind my why the meta map has been introduced here? @matthdsm? 
this is a breaking change for all pipelines doing any alignment.. it's quite a huge change, am I really the only one who's affected?

**Comment by lescai** [comment]
uhm.. I keep getting an error when I pass a dummy meta.. would be nice for modules with such a large usage to facilitate everyone’s work 😅 otherwise it’s a lot of unexpected work

**Comment by jfy133** [comment]
> uhm.. I keep getting an error when I pass a dummy meta.. would be nice for modules with such a large usage to facilitate everyone’s work sweat_smile otherwise it’s a lot of unexpected work

That's why we are retroactively going back and updating everything to have meta where possible, as this will be our general recommendation moving forward ;).

There is no way around it unfortunately (this is modern bioinformatics, things are rarely stable ;) ) ... but like I said, patch (these normally will carry over across module versions!), or just stick with the same version!

But feel free to DM me the error on slack - happy to help solve the issue :)

**Pull request merged**